packages/search/README.md

```diff
@@ -55,6 +55,113 @@ const { agoSearch } = require('@esri/hub-search')
 agoSearch(req.query, req.headers.authorization, req.headers.portal)
 ```
 
+### Content Search
+The `ContentSearchService` provides an API to search for content from the Portal API only or from both ArcGIS Online and Hub Indexer's V3 Search API (hereafter known as `V3 Search API`), depending on how the service is configured.
```

Comment:
I think how you instantiate this class in an Ember CP is really clever. 

https://github.com/ArcGIS/opendata-ui/compare/content-search-prototype#diff-3540ef806c42f464ef6f89f0d93a839089a876217c63cf6f277632ef68701f73R235-R237

That helps alleviate the concerns that I raised in https://github.com/Esri/hub.js/pull/445#discussion_r568821838

Now the CP above is on a route, and very likely we'd want to be able to search from different routes so we'd want an Ember service. I could see how we could have a `hub` service that we could inject anywhere (i.e. `@service hub`) and that in turn could then expose instances of these Hub.js "service" classes to the app as CPs like:

```js
// services/hub.js

@service appSettings;
@service session;

get contentSearchService () {
  // NOTE: does using class syntax cache this instance the way using CPs on an Ember.Object would?
  return new ContentSearchService({ portalSharingUrl: this.appSettings.sharingUrl, isPortal: this.appSettings.isPortal, userSession: this.session.authMgr });
}
```

then in some route:

```js
// some-route.js

@service hub;

searchContent (filter) {
  return this.hub.contentSearchService.search({ filter });
} 
```

Hm... after I write all that out it still feels a little cumbersome.

At the end of the day what we want is `this.hub.content.search({ filters, options})` and we're still not quite there yet, but I think that has more to do w/ the structure of Hub.js packages than what you are doing in this PR.

So, all that being said, introducing a `class` _at this time_ requires us to go through some otherwise unnecessary gymnastics. I'd much prefer if this PR just exposed the stateless fns that could later be composed into a `HubContent()` class that would then expose `.search()`, `.fetch()`, etc.

Reply:
@dbouwman  and @mjuniper what do you guys think about this?

Reply:
> Minor naming convention preference. I'm not really fond of use the `Service` suffix for classes like this and the `UserService` proposed in #445. The only reason is b/c if/when we used them in the Ember app it's going to cause confusion whether or not we are talking about an Ember service or an instance of the Hub.js class.

I'm not sure I understand this given the Intellisense IDEs have today and the prevalence of named imports.